R/app.R

```diff
@@ -212,7 +217,10 @@ shinyAppDir_serverR <- function(appDir, options=list()) {
     if (file.exists(file.path.ci(appDir, "global.R")))
       sourceUTF8(file.path.ci(appDir, "global.R"))
   }
-  onEnd <- function() {
+  onStop <- function() {
+    onStopFunc <- getShinyOption("onStop", default = function() {})
+    if (!is(onStopFunc, "function")) warning("`onStop` must be a function")
+    else onStopFunc()
```

Comment:
It's enough to just wrap `onStopFunc()` inside a `try()` call (here and in the other place), right? I did that just now and it seems to work fine (I tested by looking at the value of `getwd()`)